internal/client/api_account_test.go

```diff
@@ -0,0 +1,395 @@
+package client_test
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net/http"
+	"net/http/httptest"
+	"reflect"
+	"testing"
+
+	"github.com/Mirantis/terraform-provider-mke/internal/client"
+)
+
+type testAccountStruct struct {
+	server           *httptest.Server
+	expectedResponse client.ResponseAccount
+	expectedErr      error
+}
+
+func TestCreateValidAccount(t *testing.T) {
+	testResAcc := client.ResponseAccount{
+		ID:       "fake-test-id",
+		Name:     "testuser",
+		FullName: "Test Name",
+	}
+	mAccount, err := json.Marshal(testResAcc)
+	if err != nil {
+		t.Error(err)
+	}
+	tc := testAccountStruct{
+		server: httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+			w.WriteHeader(http.StatusOK)
+			if _, err := w.Write(mAccount); err != nil {
+				t.Error(err)
+				return
+			}
+		})),
+		expectedResponse: testResAcc,
+		expectedErr:      nil,
+	}
+	defer tc.server.Close()
+	testClient, err := client.NewUnsafeSSLClient(tc.server.URL, "fakeuser", "fakepass")
```

Comment:
consider reusing the mock testing server https://github.com/Mirantis/terraform-provider-mke/blob/main/internal/client/api_helper_test.go#L37 as the public keys test does https://github.com/Mirantis/terraform-provider-mke/blob/main/internal/client/api_publickeys_test.go#L32

Reply:
reused